src/unitxt/metrics.py

```diff
@@ -1691,16 +1765,57 @@ class F1MacroMultiLabel(F1MultiLabel):
     average = None
 
 
-class Rouge(HuggingfaceMetric):
+class Rouge(InstanceMetric):
+    main_score = "rougeL"
+    prediction_type = "str"
+    single_reference_per_prediction = False  # multiple references allowed
+    rouge_types: List[str] = ["rouge1", "rouge2", "rougeL", "rougeLsum"]
+    reduction_map = {"mean": ["rouge1", "rouge2", "rougeL", "rougeLsum"]}
+    ci_scores = ["rouge1", "rouge2", "rougeL", "rougeLsum"]
+
+    sent_split_newline: bool = True
+    _requirements_list: List[str] = ["nltk", "rouge_score"]
+
+    def compute(self, references: List[Any], prediction: Any, task_data: Dict) -> dict:
+        import nltk
+        from rouge_score import rouge_scorer
+
+        nltk.download("punkt")
+        self.sent_tokenize = nltk.sent_tokenize
```

Comment:
move to prepare

Reply:
![image](https://github.com/user-attachments/assets/9576583f-5bd3-462c-8f01-5d1cd98da02c)

Reply:
Hi @elronbandel , the above happens if I move to prepare.  is it אזעקת שוא?

Reply:
You need to save it under self.scorer then use it in the compute

Reply:
Hi @elronbandel , Done.

Reply:
@elronbandel @dafnapension - can we merge these?